src/components/trace/hat-script-graph.ts

```diff
@@ -445,14 +440,17 @@ class HatScriptGraph extends LitElement {
         </hat-graph>
         <div class="actions">
           <mwc-icon-button
-            .disabled=${paths.length === 0 || paths[0] === this.selected}
+            .disabled=${Object.keys(this.getTrackedNodes()).length === 0 ||
+            Object.keys(this.getTrackedNodes())[0] === this.selected}
             @click=${this.previousTrackedNode}
           >
             <ha-svg-icon .path=${mdiChevronUp}></ha-svg-icon>
           </mwc-icon-button>
           <mwc-icon-button
-            .disabled=${paths.length === 0 ||
-            paths[paths.length - 1] === this.selected}
+            .disabled=${Object.keys(this.getTrackedNodes()).length === 0 ||
+            Object.keys(this.getTrackedNodes())[
```

Comment:
I don't even get why you use `Object.keys()`. It returns a list ?

Reply:
Ummm... "historical reasons"?

Reply:
No, wait. It IS an object. It's the default return that is wrong.